src/app/Dashboard/DashboardLayoutUploadModal.tsx

```diff
@@ -0,0 +1,257 @@
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { FUpload, MultiFileUpload, UploadCallbacks } from '@app/Shared/FileUploads';
+import { LoadingPropsType } from '@app/Shared/ProgressIndicator';
+import { DashboardLayout, SerialCardConfig } from '@app/Shared/Redux/Configurations/DashboardConfigSlice';
+import {
+  dashboardConfigAddLayoutIntent,
+  dashboardConfigReplaceLayoutIntent,
+  RootState,
+} from '@app/Shared/Redux/ReduxStore';
+import { DashboardLayoutNamePattern } from '@app/Shared/Services/Api.service';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import { portalRoot } from '@app/utils/utils';
+import { ActionGroup, Button, Form, FormGroup, Modal, ModalVariant, Popover } from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import { forkJoin, from, Observable, of } from 'rxjs';
+import { catchError, concatMap, defaultIfEmpty, first } from 'rxjs/operators';
+
+export interface DashboardLayoutUploadModalProps {
+  visible: boolean;
+  onClose: () => void;
+}
+
+export const DashboardLayoutUploadModal: React.FC<DashboardLayoutUploadModalProps> = ({ onClose, ...props }) => {
+  const addSubscription = useSubscriptions();
+  const dispatch = useDispatch();
+  const dashboardConfigs = useSelector((state: RootState) => state.dashboardConfigs);
+  const { t } = useTranslation();
+  const submitRef = React.useRef<HTMLDivElement>(null); // Use ref to refer to submit trigger div
+  const abortRef = React.useRef<HTMLDivElement>(null); // Use ref to refer to abort trigger div
+
+  const [numOfFiles, setNumOfFiles] = React.useState(0);
+  const [allOks, setAllOks] = React.useState(false);
+  const [uploading, setUploading] = React.useState(false);
+
+  React.useEffect(() => {
+    if (allOks && numOfFiles) {
```

Comment:
This could be sending state dispatch more than needed. For example, after all succeeding, the user might remove a file from the upload list -> dispatch again.

Probably, this should be handle in the callback in `handleSubmit`. If `allOks`, then

```ts
const latest = dashboardConfigs.layouts[dashboardConfigs.layouts.length - 1];
dispatch(dashboardConfigReplaceLayoutIntent(latest.name));
```